swift/ql/lib/codeql/swift/frameworks/StandardLibrary/WebView.qll

```diff
@@ -91,39 +91,43 @@ private class JsValueSummaries extends SummaryModelCsv {
   }
 }
 
-/** The class `JSContext`. */
-private class JsContextDecl extends ClassDecl {
-  JsContextDecl() { this.getName() = "JSContext" }
+/** The `JSExport` protocol. */
+private class JsExport extends ProtocolDecl {
+  JsExport() { this.getName() = "JSExport" }
 }
 
-/** The type of an object exposed to JavaScript through `JSContext.setObject`. */
-private class TypeExposedThroughJsContext extends Type {
-  TypeExposedThroughJsContext() {
-    exists(ApplyExpr c, FuncDecl f |
-      c.getStaticTarget() = f and
-      f.getEnclosingDecl() instanceof JsContextDecl and
-      f.getName() = "setObject(_:forKeyedSubscript)"
-    |
-      c.getArgument(0).getExpr().getType() = this
-    )
-  }
+/** A protocol inheriting `JSExport`. */
+private class JsExportedProto extends ProtocolDecl {
+  JsExportedProto() { this.getABaseTypeDecl+() instanceof JsExport }
```

Comment:
Why is this not `getABaseTypeDecl*()`?  Can `JSExport` not be applied directly?

Reply:
Applying `JSExport` directly would theoretically do nothing. The expected use is extending the prototype in another prototype which declares the members you want to export to JavaScript. Then classes applying the prototype will export those members only (by default, if `JSExport` isn't used, no members are exported).